express/src/controllers/sign-in.ts

```diff
@@ -26,16 +29,45 @@ export const showLoginOtpMobile = async function (req: Request, res: Response) {
             formActionUrl: "/sign-in-otp-mobile",
             active: "sign-in",
             textMessageNotReceivedUrl: "/resend-text-code"
-
         });
     } else {
         res.redirect('/sign-in');
     }
 }
 
-export const processLoginOtpMobile = async function (req: Request, res: Response) {
-    // TO DO add the functionality to process the login mobile otp
-    if (true) { // because OTP was correct and we've implemented that
+export const processLoginOtpMobile = async function(req: Request, res: Response) {
+
+    if(!req.session.cognitoUser?.Username || !req.body['sms-otp'] || !req.session.session) {
+        res.render('there-is-a-problem.njk');
+        return;
+    }
```

Comment:
We'll probably need to think of some logging for cases like this, or move param validation to a middleware?

Reply:
100% moving stuff like this into middleware.  I just didn't want to change every method in the codebase in one PR 😆